### PR TITLE
HDDS-4449. Rename config ozone.datanode.pipeline.limit

### PR DESCRIPTION
**Pull request by wycccccc**
## What changes were proposed in this pull request?

Renamed `ozone.datanode.pipeline.limit` a suitable name.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4449

## How was this patch tested?

no test.

### CODE REVIEW

**Comment by ayushtkn** [comment]
Is directly changing the conf fine? Don't we need to deprecate the existing one? To maintain compatibility, or is that not the case here

**Comment by wycccccc** [comment]
Thanks for reminding @ayushtkn ，you are right. I will add deprecation to avoid compatibility issues.

**Comment by adoroszlai** [comment]
/pending I will add deprecation to avoid compatibility issues

**Comment by jojochuang** [comment]
Not that familiar with Ozone code but can we try using this kind of Hadoop configuration APIs?

https://github.com/apache/hadoop/blob/b69ac575a1a02d39c64c6cf998ec2ef4eb5918cd/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/HdfsConfiguration.java#L70
```
    Configuration.addDeprecations(new DeprecationDelta[]{
    new DeprecationDelta("dfs.backup.address",
    DeprecatedKeys.DFS_NAMENODE_BACKUP_ADDRESS_KEY),
```

**Comment by wycccccc** [comment]
Thanks for your suggestion @jojochuang , I found a similar method in ozone.

**Comment by wycccccc** [comment]
Thanks for your help @adoroszlai. I have corrected this, I am not sure if it is appropriate to change it here, do you think it is feasible.

**Comment by adoroszlai** [comment]
Thanks @wycccccc for adding the deprecation.  It reminded me to merge your other similar PR.  Now there is a conflict in this PR.  Sorry for the inconvenience.  Since `TestOzoneConfigurationFields` failure is caused by the changes here, a minor update is needed anyway: old config key name is leftover in the XML.

https://github.com/apache/ozone/blob/d02659c802a55366ed5d42176f83fd3917f3526c/hadoop-hdds/common/src/main/resources/ozone-default.xml#L779

**Comment by wycccccc** [comment]
Thanks @adoroszlai review, I have removed old config key and pass`TestOzoneConfigurationFields`.

**Review by adoroszlai — APPROVED** [review]
Thanks @wycccccc for updating the patch.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java [hunk 1]
```diff
@@ -293,8 +294,11 @@ public Properties getAllPropertiesByTag(String tag) {
     return configMap;
   }
 
+
```
Comment (ayushtkn) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java:297]: nit: Remove the new line change

**Review by ayushtkn — APPROVED** [review]
Thanx @wycccccc for the contribution.
Changes LGTM

**Comment by ayushtkn** [comment]
Thanx @wycccccc , @adoroszlai and @jojochuang

**Pull request merged**